### PR TITLE
Change socrata-http from 2.6 to 3.0.

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: src/main/scala/com/socrata/querycoordinator/Service.scala [hunk 1]
```diff
@@ -468,9 +475,24 @@ class Service(secondaryProvider: ServiceProviderProvider[AuxiliaryData],
...
+  private def transferHeaders(resp: HttpResponse, headers: Map[String, Seq[String]]): HttpResponse = {
+    headers.foldLeft(resp) { case (acc, (h: String, vs: Seq[String])) =>
+      vs.foldLeft(acc) { (acc2, v: String) =>
+        acc2 ~> Header(h, v)
+      }
+    }
+  }
+
+  private def resetResponse(response: HttpServletResponse): Unit = {
+    if (response.isCommitted) ???
```
Comment (sokrahta) [inline src/main/scala/com/socrata/querycoordinator/Service.scala]: not implemented

File: src/main/scala/com/socrata/querycoordinator/util/SoQLTypeCodec.scala [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
-object SoQLTypeCodec extends JsonCodec[SoQLType] {
+object SoQLTypeCodec extends JsonDecode[SoQLType] with JsonEncode[SoQLType] {
   def encode(t: SoQLType) = JString(t.name.name)
+
   def decode(v: JValue) = v match {
-    case JString(s) => SoQLType.typesByName.get(TypeName(s))
-    case _ => None
+    case JString(s) => SoQLType.typesByName.get(TypeName(s)).map(Right(_)).getOrElse(Left(InvalidValue(v)))
+    case _ => Left(InvalidValue(v))
```
Comment (sokrahta) [inline src/main/scala/com/socrata/querycoordinator/util/SoQLTypeCodec.scala]: :+1: nice to pass the decode error back to caller.

Comment (rjmac) [inline src/main/scala/com/socrata/querycoordinator/util/SoQLTypeCodec.scala]: ...thoug the "case _" should be returning an InvalidType, not an InvalidValue.

File: src/main/scala/com/socrata/querycoordinator/Service.scala [hunk 3]
```diff
@@ -389,14 +398,13 @@ class Service(secondaryProvider: ServiceProviderProvider[AuxiliaryData],
...
-            None
+            val resp = transferHeaders(Status(responseCode), headers) ~>
+              Stream(out => transferResponse(out, body))
+            Right(resp)
         }
         res match { // bit of a dance because we can't tailrec from within map
-          case Some((s, q, r)) => executeQuery(s, q, r, requestId, resourceName)
-          case None => // ok
+          case Left((s, q, r)) => executeQuery(s, q, r, requestId, resourceName, resourceScope)
+          case Right(resp) => resp
```
Comment (sokrahta) [inline src/main/scala/com/socrata/querycoordinator/Service.scala]: :+1: better control flow readability.

File: src/main/scala/com/socrata/querycoordinator/Service.scala (hunk 1 above)
Comment (chitang) [inline src/main/scala/com/socrata/querycoordinator/Service.scala]: Not implemented is not new.  It is carried over from  if(resp.isCommitted) line 436.

**Comment by sokrahta** [comment]
lgtm :shipit:

**Pull request merged**